orc/src/main/java/org/apache/iceberg/orc/OrcIterable.java

```diff
@@ -84,15 +91,18 @@ public CloseableIterator<T> iterator() {
     addCloseable(orcFileReader);
 
     TypeDescription fileSchema = orcFileReader.getSchema();
+    Schema schemaWithoutConstantFields =
```

Comment:
Previously, schema already pruned without constant columns is passed into ORCIterable for its filter logic.

However, this fails attempts to bind constant column filters.  This makes it so that we prune it only when we need it, but we keep it in attempting to bind the constant column filters.